### DISCUSSION TITLE
How to produce messages with avro serde?..

### DISCUSSION THREAD

**Discussion by HyundeokPark** [provectus/kafka-ui#3355] [Q&A] [answered]
Hi, I'm trying to produce messages with avro serde, but there is no option for avro like this,
<img width="1152" alt="image" src="https://user-images.githubusercontent.com/12470341/219284236-5d411899-341f-40b6-9667-223b9ee9e9df.png">
<img width="1153" alt="image" src="https://user-images.githubusercontent.com/12470341/219284272-3f74a252-6eef-40c4-9f22-80b30bff9f76.png">

Currently I'm using docket images with tag 'master' and in Topics - Messages section, I see Key/Value Serde have SchemaRegistry option
<img width="1205" alt="image" src="https://user-images.githubusercontent.com/12470341/219284143-adee9288-ebaf-4524-89d2-4ed9c2746668.png">

when i produce messages with String serde, my connectors are into failed task with
 error below...

 `
org.apache.kafka.connect.errors.ConnectException: Tolerance exceeded in error handler at org.apache.kafka.connect.runtime.errors.RetryWithToleranceOperator.execAndHandleError(RetryWithToleranceOperator.java:223) at org.apache.kafka.connect.runtime.errors.RetryWithToleranceOperator.execute(RetryWithToleranceOperator.java:149) at org.apache.kafka.connect.runtime.WorkerSinkTask.convertAndTransformRecord(WorkerSinkTask.java:516) at org.apache.kafka.connect.runtime.WorkerSinkTask.convertMessages(WorkerSinkTask.java:493) at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:332) at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:234) at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:203) at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:189) at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:244) at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) at java.base/java.lang.Thread.run(Thread.java:833) Caused by: org.apache.kafka.connect.errors.DataException: Failed to deserialize data for topic labeled to Avro: at io.confluent.connect.avro.AvroConverter.toConnectData(AvroConverter.java:124) at org.apache.kafka.connect.storage.Converter.toConnectData(Converter.java:88) at org.apache.kafka.connect.runtime.WorkerSinkTask.lambda$convertAndTransformRecord$4(WorkerSinkTask.java:516) at org.apache.kafka.connect.runtime.errors.RetryWithToleranceOperator.execAndRetry(RetryWithToleranceOperator.java:173) at org.apache.kafka.connect.runtime.errors.RetryWithToleranceOperator.execAndHandleError(RetryWithToleranceOperator.java:207) ... 13 more Caused by: org.apache.kafka.common.errors.SerializationException: Unknown magic byte! at io.confluent.kafka.serializers.AbstractKafkaSchemaSerDe.getByteBuffer(AbstractKafkaSchemaSerDe.java:250) at io.confluent.kafka.serializers.AbstractKafkaAvroDeserializer$DeserializationContext.<init>(AbstractKafkaAvroDeserializer.java:334) at io.confluent.kafka.serializers.AbstractKafkaAvroDeserializer.deserializeWithSchemaAndVersion(AbstractKafkaAvroDeserializer.java:202) at io.confluent.connect.avro.AvroConverter$Deserializer.deserialize(AvroConverter.java:172) at io.confluent.connect.avro.AvroConverter.toConnectData(AvroConverter.java:107) ... 17 more
`

is there somethin I missed?... 
Thank you for yout answers in advance

P.S 
I use strimzi for provisioning kafka, and kafka connect, connector plugins, also schema registry is provisioned by [bitnami](https://bitnami.com/stack/schema-registry)

**Comment by Amaralus** [comment]
hi, i have a same issue

**Reply by HyundeokPark** [reply]
I have figured it out my self, I didn't know that there are some rules 

when you named schema subject name,

it should be  form of `topic's name-value` or `topic's name-key` 
<img width="787" alt="image" src="https://user-images.githubusercontent.com/12470341/219569048-9d232cbe-cc9e-4acf-a797-e270608edb7b.png">


if i have topic name 'label', then i should named 'label-value' for value serde

and same for the key

and this is the results I got

<img width="1149" alt="image" src="https://user-images.githubusercontent.com/12470341/219568926-6a30c496-5f85-465d-9354-3d525fb895c7.png">

**Reply by iliax** [reply, marked as answer]
@HyundeokPark yes, you are right 
kafka-ui only supports `TopicNameStrategy` for serialization now (see [doc](https://docs.confluent.io/platform/current/schema-registry/serdes-develop/index.html#overview)).
We are planning to add other policies in the future

**Comment by javapapo** [comment]
Hello not sure if this is related but I am currently running 

v0.6.1

And I am still not getting Avro msg properly preview-ed

<img width="1485" alt="image" src="https://user-images.githubusercontent.com/484186/231729226-c8845057-3d2d-4880-ac58-a1510205ff72.png">

**Reply by iliax** [reply]
Hello @javapapo ,
Are you using SchemaRegistry for avro schemas storage? If so - you need to add SR-related properties to your configuration. (see [basic config ](https://docs.kafka-ui.provectus.io/configuration/configuration-file#basic-config-structure) for sample.

When configured properly - you will be able to select "SchemaRegistry" serde list. 

"Avro (Embedded)" serde is another type of serialization - when schema is embedded in message payload.

**Reply by javapapo** [reply]
Many thanks for the prompt reply - apologies I am slightly lost.

Do you mean this one?

```
schemaNameTemplate: how keys are saved to Schema Registry
```
I do have weird the SR to the basic config .

```    
KAFKA_CLUSTERS_0_SCHEMA_REGISTRY: "https://xxxxxxxx-internal-eks.eu-xxxxxx"
```

And the integration is working..as far as I understand

<img width="1870" alt="image" src="https://user-images.githubusercontent.com/484186/231732389-b0dd1dcf-4668-4e39-a64d-a6847d7d251a.png">

**Reply by iliax** [reply]
@javapapo ok, 
1. do you have "Schema Registry" line in left menu?
2. do you have "SchemaRegistry" serde option Value Serde dropdown (on your topic page)?

**Comment by javapapo** [comment]
1. Yes
<img width="164" alt="image" src="https://user-images.githubusercontent.com/484186/231753371-8760a85f-93ef-49bf-81d5-be25a21d0972.png">

2. No I only get this

<img width="1328" alt="image" src="https://user-images.githubusercontent.com/484186/231753806-98639a90-0112-4746-9778-f51a5f5acba8.png">

And this is how it looks like from the Schema Registry side
![image](https://user-images.githubusercontent.com/484186/231754648-4de3920c-cbc8-44bf-b2a6-4357e3f292e8.png)

Thanks for taking the time :)

**Reply by iliax** [reply]
@javapapo ok, I got it 
can you please set `KAFKA_CLUSTERS_0_SCHEMAREGISTRY ` property instead of `KAFKA_CLUSTERS_0_SCHEMA_REGISTRY` ? It should fix the issue

**Reply by javapapo** [reply]
Ooh! let me try!

**Reply by javapapo** [reply]
Legend! @iliax many many thanks for your time! Highly Appreciated!
